model/event/auth_event.go

```diff
@@ -0,0 +1,92 @@
+package event
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+)
+
+const (
+	ResourceTypeUserRoles = "USER_ROLES"
+	ResourceTypeRole      = "ROLE"
+	EventTypeUserRoles    = "USER_ROLES_CHANGED"
+	EventTypeRole         = "ROLE_CHANGED"
+)
+
+type RoleChangeOperation string
+
+const (
+	AddRole    RoleChangeOperation = "add"
+	UpdateRole RoleChangeOperation = "update"
+	RemoveRole RoleChangeOperation = "remove"
+)
+
+func (op RoleChangeOperation) validate() error {
+	switch op {
+	case AddRole, UpdateRole, RemoveRole:
+		return nil
+	default:
+		return errors.Errorf("invalid role change operation '%s'", op)
+	}
+}
+
+type userRolesData struct {
+	User      string              `bson:"user" json:"user"`
+	RoleID    string              `bson:"role_id" json:"role_id"`
+	Operation RoleChangeOperation `bson:"operation" json:"operation"`
+}
+
+func LogUserRolesEvent(user, roleID string, op RoleChangeOperation) error {
```

Comment:
This should be more generic, so that it logs any change to a user. The existing types of changes can be the only ones supported for now